sa/model.go

```diff
@@ -12,23 +12,137 @@ import (
 	"github.com/letsencrypt/boulder/core"
 	"github.com/letsencrypt/boulder/features"
 	"github.com/letsencrypt/boulder/probs"
+	"github.com/letsencrypt/boulder/revocation"
 )
 
-const (
-	regV1Fields        string = "id, jwk, jwk_sha256, contact, agreement, initialIP, createdAt, LockCol"
-	regV2Fields        string = regV1Fields + ", status"
-	pendingAuthzFields string = "id, identifier, registrationID, status, expires, combinations, LockCol"
-	authzFields        string = "id, identifier, registrationID, status, expires, combinations"
-	sctFields          string = "id, sctVersion, logID, timestamp, extensions, signature, certificateSerial, LockCol"
+type dbSelectOne func(interface{}, string, ...interface{}) error
+type dbSelect func(interface{}, string, ...interface{}) ([]interface{}, error)
 
-	// CertificateFields and CertificateStatusFields are also used by cert-checker and ocsp-updater
-	CertificateFields       string = "registrationID, serial, digest, der, issued, expires"
-	CertificateStatusFields string = "serial, subscriberApproved, status, ocspLastUpdated, revokedDate, revokedReason, lastExpirationNagSent, ocspResponse, LockCol"
+const regFields = "id, jwk, jwk_sha256, contact, agreement, initialIP, createdAt, LockCol"
+const regFieldsv2 = regFields + ", status"
 
-	// CertificateStatusFieldsv2 is used when the CertStatusOptimizationsMigrated
-	// feature flag is enabled and includes "notAfter" and "isExpired" fields
-	CertificateStatusFieldsv2 string = CertificateStatusFields + ", notAfter, isExpired"
-)
+func SelectRegistration(so dbSelectOne, q string, args ...interface{}) (*regModelv1, error) {
```

Comment:
To satisfy go vet (which we don't run automatically), each of these exported functions should have a comment explaining what they do and why.